### PR TITLE
[DRAFT] HTTP Examples

### PR DESCRIPTION
**Pull request by PineappleIOnic**
This PR adds examples for HTTP into the appwrite SDK Generator. It still needs alot of work and this PR is mainly open to continue conversation on what needs to be modified and improved with the examples.

Thank you for reviewing this PR

### CODE REVIEW

File: src/SDK/Language/HTTP.php [hunk 1]
```diff
@@ -172,12 +173,12 @@ public function getParamExample(array $param)
...
-            [
-                'scope'         => 'method',
-                'destination'   => 'docs/examples/{{service.name | caseLower}}/{{method.name | caseDash}}.md',
-                'template'      => 'http/docs/example.md.twig',
-                'minify'        => false,
-            ],
+          [
+            'scope'         => 'method',
+            'destination'   => 'docs/examples/{{service.name | caseLower}}/{{method.name | caseDash}}',
+            'template'      => '/http/docs/service.twig',
```
Comment (eldadfux) [inline src/SDK/Language/HTTP.php:179]: example.md.twig will be more consistent use of the file name compared to other language templates.

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: templates/http/docs/service.twig [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{% endif %}
+{% endfor %}
+
+{% for key, header in method.headers %}
+{% if header == 'application/json' %}
+{% if method.parameters.body %}
+json: {
+{% for parameter in method.parameters.body %}
+  "{{ parameter.name }}": {{ parameter | paramExample }}{% if not loop.last %},{% endif %}
+
```
Comment (lohanidamodar) [inline templates/http/docs/service.twig:22]: Is this extra spacing required here?

File: templates/http/docs/service.twig [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{% endif %}
+{% endif %}
+{% if header == 'multipart/form-data' %}
+{% set boundary = random() %}
+boundary={{boundary}}
+Content-Length: *Length of your entity body in bytes*
+{% for parameter in method.parameters.body %}
+--{{boundary}}
+Content-Disposition: form-data; name="{{parameter.name}}{% if parameter.type == "array" %}[]{% endif %}"
+{% if parameter.type != 'file' %}{% endif %}
```
Comment (lohanidamodar) [inline templates/http/docs/service.twig:34]: `if` and `endif` together without body? What's this doing?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Great Job @PineappleIOnic.
Just some extra spacing and some things I don't understand.

File: templates/http/docs/service.twig (hunk 3 above)
Comment (PineappleIOnic) [reply templates/http/docs/service.twig:34]: Whoops, that was left there from when I was working on the file metadata but it wasn't needed for formdata

Comment (PineappleIOnic) [reply templates/http/docs/service.twig:34]: I'll remove it now

File: templates/http/docs/service.twig (hunk 2 above)
Comment (PineappleIOnic) [reply templates/http/docs/service.twig:22]: Without it the JSON gets formatted wierdly like so:
```json
json: {
"email": email@example.com,"password": password}
```
with it, it'll get formatted correctly like so: 
```json
json: {
  "email": email@example.com,
  "password": password
}
```
Correctly adding indentation and new lines, if I'm missing a better way of doing it please let me know 👍

**Review by lohanidamodar — APPROVED** [review]

**Comment by eldadfux** [comment]
Awesome work @PineappleIOnic 🎉

**Pull request merged**